lib/hsa/hsaRfiUpdateBadInputs.cpp

```diff
@@ -87,64 +111,21 @@ hsa_signal_t hsaRfiUpdateBadInputs::execute(int gpu_frame_id, hsa_signal_t prece
 
 void hsaRfiUpdateBadInputs::finalize_frame(int frame_id) {
     std::lock_guard<std::mutex> lock(update_mutex);
-    if (update_bad_inputs && frames_to_update_finalize > 0) {
-        frames_to_update_finalize--;
+    
+    // Only mark input empty if filling frame and
+    // no more frames to finalize.
+    if (frame_to_fill_finalize > 0 && frame_copy_active.at(frame_id)) {
```

Comment:
We don't actually need the `frame_to_fill_finalize` condition anymore.  The `frame_copy_active.at(frame_id)` is sufficient as far as I can tell.